src/components/pages/devices/flyouts/deviceJobs/deviceJobTags.js

```diff
@@ -1,20 +1,302 @@
 // Copyright (c) Microsoft. All rights reserved.
 
 import React from 'react';
+import { Link } from "react-router-dom";
+import { Observable } from 'rxjs';
+import update from 'immutability-helper';
 
+import { IoTHubManagerService } from 'services';
+import { toSubmitTagsJobRequestModel } from 'services/models';
+import { LinkedComponent } from 'utilities';
+import { svgs, Validator } from 'utilities';
 import {
+  AjaxError,
+  Btn,
+  BtnToolbar,
+  ErrorMsg,
+  FormControl,
+  FormGroup,
+  FormLabel,
   FormSection,
+  Indicator,
   SectionDesc,
-  SectionHeader
-} from '../../../../shared';
-
-export const DeviceJobTags = (props) => {
-  const { t, devices } = props;
-  return (
-    <FormSection className="device-job-tags-container">
-      <SectionHeader>{t('devices.flyouts.jobs.tags.title')}</SectionHeader>
-      <SectionDesc>{t('devices.flyouts.jobs.tags.description')}</SectionDesc>
-      <pre>{JSON.stringify(devices, null, 2)}</pre>
-    </FormSection>
-  )
+  SectionHeader,
+  SummaryCount,
+  SummarySection,
+  Svg
+} from 'components/shared';
+import {
+  PropertyGrid as Grid,
+  PropertyGridBody as GridBody,
+  PropertyGridHeader as GridHeader,
+  PropertyRow as Row,
+  PropertyCell as Cell
+} from 'components/pages/devices/flyouts/deviceDetails/propertyGrid';
+
+update.extend('$autoArray', function(value, object) {
+  return object ?
+    update(object, value):
+    update([], value);
+});
+
+const isNumeric = value => typeof value === 'number' || !isNaN(parseInt(value, 10));
+const isAlphaNumericRegex = /^[a-zA-Z0-9]*$/;
+const nonAlphaNumeric = x => !x.match(isAlphaNumericRegex);
+
+const TagJobConstants = {
```

Comment:
camelCase

Reply:
camelCase would be inconsistent with previous PRs with similar exterrnal entities... see deviceNew.js, iotHubManagerModels.js